### PR TITLE
Functional tests - Product settings separator of attribute anchor on the product links

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Test separator of attribute anchor on the product links.
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/13_shopParameters/03_productSettings/02_productPage/05_separatorOfAttributeAnchor.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18174)

### CODE REVIEW

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/05_separatorOfAttributeAnchor.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+require('module-alias/register');
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_productSettings_separatorOfAttributeAnchorOnProductLinks';
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/05_separatorOfAttributeAnchor.js]:
```suggestion
const baseContext = 'functional_BO_productSettings_separatorOfAttributeAnchor';
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/05_separatorOfAttributeAnchor.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+const ProductPage = require('@pages/FO/product');
+const HomePage = require('@pages/FO/home');
+const SearchResultsPage = require('@pages/FO/searchResults');
+// Importing data
+const {Products} = require('@data/demo/products');
+
+let browser;
+let page;
+const attributesWithSeparator2 = '1,size,s/8,color,white';
+const attributesWithSeparator1 = '1-size-s/8-color-white';
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/05_separatorOfAttributeAnchor.js]:
```suggestion
const attributesWithSeparator = ['1', 'size', 's/8', 'color', 'white'];
```
Just a suggestion. See what I suggested below for your `forEach`.

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/05_separatorOfAttributeAnchor.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      this.pageObjects.boBasePage.shopParametersParentLink,
+      this.pageObjects.boBasePage.productSettingsLink,
+    );
+    const pageTitle = await this.pageObjects.productSettingsPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.productSettingsPage.pageTitle);
+  });
+
+  const tests = [
+    {args: {option: ',', attributesInProductLink: attributesWithSeparator2}},
+    {args: {option: '-', attributesInProductLink: attributesWithSeparator1}},
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/05_separatorOfAttributeAnchor.js]: You can use what I described earlier here:
```suggestion
    {args: {option: ',', attributesInProductLink: attributesWithSeparator.join(',')}},
    {args: {option: '-', attributesInProductLink: attributesWithSeparator.join('-')}},
```
You could even make it automatic with a second array providing the separators.

**Review by SimonGrn — COMMENTED** [review]
thanks @nesrineabdmouleh

**Review by SimonGrn — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Pull request merged**